llvm/lib/Target/NVPTX/NVPTXInstrInfo.td

```diff
@@ -1193,6 +1206,10 @@ defm FNEG_H: F2_Support_Half<"neg", fneg>;
 
 defm FSQRT : F2<"sqrt.rn", fsqrt>;
 
+defm FEXP2 : F1<"ex2.approx", fexp2>;
```

Comment:
Looks like we already have these instructions here: 

https://github.com/llvm/llvm-project/blob/fd784726db70a5155594c32ee839b8807fafd87d/llvm/lib/Target/NVPTX/NVPTXIntrinsics.td#L1252-L1268

Instead of creating a duplicate set, can you use patterns to rewrite these nodes to those existing instructins?

Reply:
Good catch. And I've added support for `ex2.approx.{f16,f16x2,bf16,bf16x2}` variants.